front-end/src/pages/PostDetailPage.tsx

```diff
@@ -0,0 +1,62 @@
+import axios, { AxiosResponse } from 'axios';
+import React, { useEffect, useState } from 'react';
+import { RouteComponentProps } from 'react-router';
+import { useParams } from 'react-router-dom';
+import { Row, Col } from 'reactstrap';
+import PostDetailModal from '../components/PostDetailModal';
+import { Post } from '../types/Post';
+import { UserLogin } from '../types/UserLogin';
+
+
+interface Props extends RouteComponentProps<MatchParams>{
+  loggedInUser: UserLogin|undefined;
+}
+
+interface MatchParams {
+  postId: string
+}
+
+/**
+ * Renders a post detail
+ * @param props 
+ */
+export default function PostDetailPage(props: Props) {
+  const [postEntry, setPostEntry] = useState<Post | undefined>(undefined);
+
+  const {postId} = props.match.params;
+  // get post
+  useEffect(() => {
+    let getPromise: Promise<AxiosResponse<any>>;
+    if (props.loggedInUser) {
+      getPromise = axios.get(`${process.env.REACT_APP_API_URL}/api/posts/${postId}`,
+        {
+          auth: {
+            username: props.loggedInUser.username,
+            password: props.loggedInUser.password
+          }
+        });
+    }
+    else {
+      getPromise = axios.get(`${process.env.REACT_APP_API_URL}/api/posts/${props.location.pathname}`,);
```

Comment:
Just to make sure I understand 100%, what's the difference between making the request w/ Basic Auth and w/o?

Reply:
Good question :+1:

This logic is to allow for logged-in users and non-logged-in users to attempt to access a post.
I'm assuming that some friends-only posts need user credentials in order to determine if the server should return the post or not

I'm assuming that the server will refuse to send a friends-only post if 'bad' credentials or no credentials get sent to it

There might need to be additional work to get this edge case completed though

Reply:
Got it, makes sense. Thanks for clarifying!